src/librepgp/stream-common.h

```diff
@@ -353,4 +353,11 @@ void *mem_dest_own_memory(pgp_dest_t *dst);
  **/
 rnp_result_t init_null_dest(pgp_dest_t *dst);
 
+/** @brief reads from source and writes to destination
+ *  @param src initialized source
+ *  @param dst initialized destination
+ *  @return RNP_SUCCESS or error code
+ **/
+rnp_result_t rnp_pipe_source(pgp_source_t *src, pgp_dest_t *dst);
```

Comment:
At this layer no need to use `rnp_` prefix. This could be better called `src_pipe()`, `dst_write_src()` or so on.